### PR TITLE
Auth branch

### PR DESCRIPTION
**Pull request by anastasiya-charnahayeva**
auth block

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,9 +18,12 @@
     ]
   },
   "dependencies": {
+    "firebase": "^10.12.2",
     "lint-staged": "^15.2.5",
```
Comment (ghost) [inline package.json:22]: All of the dependencies which are listed in "Dependencies" are included in a bundle. So please pay attention to exclude dev runtime only to "DevDependencies"
`lint-staged` is only used during commit time, so it should be in dev ones

File: src/App.vue [hunk 2]
```diff
@@ -1,30 +1,27 @@
-<script setup lang="ts">
-import HelloWorld from './components/HelloWorld.vue'
-</script>
+<script setup lang="ts"></script>
 
 <template>
   <div>
```
Comment (ghost) [inline src/App.vue:4]: An empty divs again 👀

File: src/App.vue [hunk 3]
```diff
@@ -1,30 +1,27 @@
...
+  import { useAuthStore } from "./stores/modules/useAuth";
+  import { onAuthStateChanged } from "firebase/auth";
+  import fbAuth from "./firebase/index";
+  import Header from "./components/Header.vue";
+
+  const authStore = useAuthStore();
+  const { setCurrentUser } = authStore;
+  const { user } = storeToRefs(authStore);
+
+  onMounted(() => {
```
Comment (ghost) [inline src/App.vue:22]: This one is nice to be a part of auth store and not directly inside of a component

File: src/firebase.config.js
Comment (ghost) [inline src/firebase.config.js:1]: In this the same file with /index.js inside of this directory. By the way extension is .js and not a .ts

File: src/stores/modules/useAuth.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  getAuth,
+  sendEmailVerification,
+  signInWithEmailAndPassword,
+  signOut,
+} from "firebase/auth";
+
+export const useAuthStore = defineStore(
+  "auth",
+  () => {
+    const user = ref();
```
Comment (ghost) [inline src/stores/modules/useAuth.ts:15]: I think there should be some typescript errors here

File: src/views/Login.vue [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  import { reactive, ref } from "vue";
+  import { useRouter } from "vue-router";
+
+  const form = reactive({
+    email: "",
+    password: "",
+  });
+
+  const show_alert = ref<boolean>(false);
+  const alert_variant = ref<string>("bg-blue-500");
```
Comment (ghost) [inline src/views/Login.vue:80]: Inconsistent naming again 👀  Don't change for now, but please pay attention

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/stores/modules/useAuth.ts (hunk 4 above)
Comment (anastasiya-charnahayeva) [reply src/stores/modules/useAuth.ts:15]: ?

**Pull request merged**